docker-compose.yml

```diff
@@ -0,0 +1,20 @@
+version: '3.1'
+
+services:
+  test:
+    build: .
+    command: ["go", "test", "-v", "./..."]
+    depends_on:
+      - postgres
+    environment:
+      DATABASE_URL: postgres://pguser:pguser@postgres:5432/pgdb
+      RUN_MIGRATIONS: "YES"
+  postgres:
+    image: postgres
+    restart: always
+    ports:
+      - "5432:5432"
+    environment:
+      POSTGRES_USER: pguser
```

Comment:
kinda nitpicky, but this `POSTGRES_USER` value is [already the default](https://github.com/docker-library/postgres/blob/aaf209829e072bec11bca1f6f0b52e756dfc096e/10/docker-entrypoint.sh#L101). additionally, the image sets `POSTGRES_PASSWORD` to the value of `POSTGRES_USER`.

one idea: add an `.env` file, set the values for `POSTGRES_USER|PASSWORD|DB` to something explicit (e.g., `migrate`), and use those values in your `DATABASE_URL` above, e.g., `postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}:5432/${POSTGRES_DB}`.

Reply:
I like the simplicity and transparency here and don't really wanna add yet another top level configuration file that's only used for running the tests to avoid copy and pasting 3 values. I like the idea of changing the variables away from their defaults though, going to push a commit to do so.